test/py/test_device_get_region_io_fds.py

```diff
@@ -33,149 +31,232 @@ def test_device_get_region_io_fds_setup():
+    payload = vfio_user_region_io_fds_request(\
+    argsz = ctypes.sizeof(vfio_user_region_io_fds_reply) + \
+        ctypes.sizeof(vfio_user_sub_region_ioeventfd) * 5, flags = 1, \
+                                index = VFU_PCI_DEV_BAR2_REGION_IDX, count = 0)
 
-    msg(ctx, sock, VFIO_USER_DEVICE_GET_REGION_IO_FDS, payload, expect=errno.EINVAL)
+    msg(ctx, sock, VFIO_USER_DEVICE_GET_REGION_IO_FDS, payload, \
+        expect=errno.EINVAL)
 
 def test_device_get_region_io_fds_bad_count():
 
-    payload = region_io_fds_request(argsz = 16+40*5, flags = 0, index = VFU_PCI_DEV_BAR2_REGION_IDX, count = 1)
+    payload = vfio_user_region_io_fds_request(\
+    argsz = ctypes.sizeof(vfio_user_region_io_fds_reply) + \
+        ctypes.sizeof(vfio_user_sub_region_ioeventfd) * 5, flags = 0, \
+                                index = VFU_PCI_DEV_BAR2_REGION_IDX, count = 1)
+
+    msg(ctx, sock, VFIO_USER_DEVICE_GET_REGION_IO_FDS, payload, \
+        expect=errno.EINVAL)
+
+def test_device_get_region_io_fds_buffer_too_small():
+
+    payload = vfio_user_region_io_fds_request(\
+        argsz = ctypes.sizeof(vfio_user_region_io_fds_reply) -1 , flags = 0, \
+                                index = VFU_PCI_DEV_BAR2_REGION_IDX, count = 1)
+
+    msg(ctx, sock, VFIO_USER_DEVICE_GET_REGION_IO_FDS, payload, \
+        expect=errno.EINVAL)
+
+def test_device_get_region_io_fds_buffer_too_large():
+
+    payload = vfio_user_region_io_fds_request(argsz = SERVER_MAX_DATA_XFER_SIZE\
+                                              + 1, flags = 0, \
+                                index = VFU_PCI_DEV_BAR2_REGION_IDX, count = 1)
+
+    msg(ctx, sock, VFIO_USER_DEVICE_GET_REGION_IO_FDS, payload, expect = \
+        errno.EINVAL)
+
+def test_device_get_region_io_fds_no_regions():
+
+    payload = vfio_user_region_io_fds_request(argsz = 512, flags = 0, \
+                                index = VFU_PCI_DEV_BAR1_REGION_IDX, count = 0)
+
+    ret = msg(ctx, sock, VFIO_USER_DEVICE_GET_REGION_IO_FDS, payload, expect=0)
+
+    reply, ret = vfio_user_region_io_fds_reply.pop_from_buffer(ret)
+
+    assert reply.argsz == ctypes.sizeof(vfio_user_region_io_fds_reply)
+    assert reply.count == 0
+    assert reply.flags == 0
+    assert reply.index == VFU_PCI_DEV_BAR1_REGION_IDX
 
-    msg(ctx, sock, VFIO_USER_DEVICE_GET_REGION_IO_FDS, payload, expect=errno.EINVAL)
 
-def test_device_get_region_io_fds_buffer_to_small():
+def test_device_get_region_io_fds_no_regions_setup():
 
-    payload = region_io_fds_request(argsz = 15, flags = 0, index = VFU_PCI_DEV_BAR2_REGION_IDX, count = 1)
+    payload = vfio_user_region_io_fds_request(argsz = 512, flags = 0, \
+                                index = VFU_PCI_DEV_BAR3_REGION_IDX, count = 0)
 
-    msg(ctx, sock, VFIO_USER_DEVICE_GET_REGION_IO_FDS, payload, expect=errno.EINVAL)
+    ret = msg(ctx, sock, VFIO_USER_DEVICE_GET_REGION_IO_FDS, payload, expect=0)
```

Comment:
should we really get  errno of 0, if the region isn't even set up? Probably not right?